GITHUB-5937 - Remove mcrypt requirement

BatsaxIV opened the pull request:
**Description (for Contributor and Core Developer)**
Fix https://github.com/akeneo/pim-community-dev/issues/5937 by removing mcrypt requirement.

After searching any mcrypt function on the project, it does not seem to be used (as stated by @gquemener and @a2xchip).

I've added a **Requirements** part in the Changelog, not really sure where to do add this modification, is this correct?

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

BatsaxIV commented:
If this is ok, it needs a change on the [documentation](https://docs.akeneo.com/latest/developer_guide/installation/system_requirements/system_requirements.html) (I'll do it).

nidup commented:
Hi @BatsaxIV 

Thank you for the pull request! :tada: 

I look at the code too and as you, i don't find real usage of mcrypt functions.

I only found references in vendor code we don't directly use in the PIM,

```$ grep -r 'mcrypt' ./
./vendor/paragonie/random_compat/CHANGELOG.md:     * Don't unnecessarily prevent `mcrypt_create_iv()` from being used.
./vendor/paragonie/random_compat/CHANGELOG.md:  accessible to PHP, skip mcrypt (which just fails before giving OpenSSL
./vendor/paragonie/random_compat/CHANGELOG.md:* PHP 5.6.10 - 5.6.12 will hang when mcrypt is used on Unix-based operating 
./vendor/paragonie/random_compat/CHANGELOG.md:support for PHP < 5.4.1 without mcrypt on Windows 7+ in the process. Since this
./vendor/paragonie/random_compat/CHANGELOG.md:There is an outstanding issue `mcrypt_create_iv()` and PHP 7's `random_bytes()`
./vendor/paragonie/random_compat/CHANGELOG.md:* Prioritize `/dev/arandom` and `/dev/urandom` over mcrypt.
./vendor/paragonie/random_compat/CHANGELOG.md:    1. `mcrypt_create_iv()` with `MCRYPT_DEV_URANDOM`
./vendor/paragonie/random_compat/RATIONALE.md: 3. `mcrypt_create_iv($bytes, MCRYPT_DEV_URANDOM)`
./vendor/paragonie/random_compat/RATIONALE.md:Despite [strongly urging people not to use mcrypt in their projects](https://paragonie.com/blog/2015/05/if-you-re-typing-word-mcrypt-into-your-code-you-re-doing-it-wrong)
./vendor/paragonie/random_compat/RATIONALE.md:(because libmcrypt is abandonware and the API puts too much responsibility on the
./vendor/paragonie/random_compat/RATIONALE.md:implementor) we prioritize `mcrypt_create_iv()` with `MCRYPT_DEV_URANDOM` above
./vendor/paragonie/random_compat/RATIONALE.md:The reason is simple: `mcrypt_create_iv()` is part of PHP's `ext/mcrypt` code,
./vendor/paragonie/random_compat/RATIONALE.md:and is not part `libmcrypt`. It actually does the right thing:
./vendor/paragonie/random_compat/RATIONALE.md:If we're on Windows and don't have access to `mcrypt`, we use `CAPICOM.Utilities.1`.
./vendor/paragonie/random_compat/lib/random.php:     *   3. mcrypt_create_iv($bytes, MCRYPT_DEV_URANDOM)
./vendor/paragonie/random_compat/lib/random.php:     * mcrypt_create_iv()
./vendor/paragonie/random_compat/lib/random.php:     * - the mcrypt extensions is loaded
./vendor/paragonie/random_compat/lib/random.php:     *     5.6.10 and 5.6.12, we don't want to use mcrypt. It will
./vendor/paragonie/random_compat/lib/random.php:        extension_loaded('mcrypt')
./vendor/paragonie/random_compat/lib/random.php:        // See random_bytes_mcrypt.php
./vendor/paragonie/random_compat/lib/random.php:        require_once $RandomCompatDIR . '/random_bytes_mcrypt.php';
./vendor/paragonie/random_compat/lib/random.php:     * This is a Windows-specific fallback, for when the mcrypt extension
./vendor/paragonie/random_compat/lib/random_bytes_mcrypt.php:     * Powered by ext/mcrypt (and thankfully NOT libmcrypt)
./vendor/paragonie/random_compat/lib/random_bytes_mcrypt.php:     * @ref https://github.com/php/php-src/blob/c568ffe5171d942161fc8dda066bce844bdef676/ext/mcrypt/mcrypt.c#L1321-L1386
./vendor/paragonie/random_compat/lib/random_bytes_mcrypt.php:        $buf = @mcrypt_create_iv($bytes, MCRYPT_DEV_URANDOM);
./vendor/composer/installed.json:            "ext-mcrypt": "*"
./vendor/ircmaxell/password-compat/lib/password.php:                if (function_exists('mcrypt_create_iv') && !defined('PHALANGER')) {
./vendor/ircmaxell/password-compat/lib/password.php:                    $buffer = mcrypt_create_iv($raw_salt_len, MCRYPT_DEV_URANDOM);
./vendor/sebastian/resource-operations/src/ResourceOperations.php:            'mcrypt_module_open',
./vendor/swiftmailer/swiftmailer/CHANGES: * removed the need to have mcrypt installed
./vendor/ass/xmlsecurity/composer.json:        "ext-mcrypt": "*"
./vendor/ass/xmlsecurity/src/ass/XmlSecurity/Key/Block.php:            // for PHP 5.3 fall back to mcrypt if available
./vendor/ass/xmlsecurity/src/ass/XmlSecurity/Key/Block.php:            } elseif (function_exists('mcrypt_decrypt')) {
./vendor/ass/xmlsecurity/src/ass/XmlSecurity/Key/Block.php:                    $mcryptCipherMethod = MCRYPT_TRIPLEDES;
./vendor/ass/xmlsecurity/src/ass/XmlSecurity/Key/Block.php:                    $mcryptCipherMethod = MCRYPT_RIJNDAEL_128;
./vendor/ass/xmlsecurity/src/ass/XmlSecurity/Key/Block.php:                $decryptedData = mcrypt_decrypt($mcryptCipherMethod, $this->key, $data, MCRYPT_MODE_CBC, $iv);
./vendor/ass/xmlsecurity/src/ass/XmlSecurity/Key/Block.php:     * http://us3.php.net/manual/en/function.mcrypt-encrypt.php#102428
```

To totally remove the dep, you also need to drop it from https://github.com/akeneo/pim-community-dev/blob/master/app/OroRequirements.php#L42

Once this done, i will re-run a full build and merge the pull request if everything goes well.

Thank you!

BatsaxIV commented:
Hi @nidup,

I've already removed the lines in the OroRequirements.php file, see [here](https://github.com/akeneo/pim-community-dev/pull/6081/files#diff-1e85c1e53ead00ee5b86cd2ce99dfc6dL43).
It's just the _diff_ that has no difference for l.42, which is the case ; and so, removed the l.48 instead.

Ok for you?

BatsaxIV commented:
Should I change the modification on the changelog to 1.8.x file ?

nidup commented:
Exact! thank you to move it to the CHANGELOG-1.8.md :+1:

BatsaxIV commented:
Done ;)

nidup commented:
Thank you! :champagne:

nidup commented:
You can claim the core contributor badge on badger :dancer: http://badger.akeneo.com/badge/41acec2c-649f-11e6-92dc-d60437e930cf
